native/src/multi_sig.rs

```diff
@@ -0,0 +1,360 @@
+    object
+        .set(&mut ctx, "partialSignature", partial_signature_buffer)
+        .chain_neon(&mut ctx, "Unable to create object with key partialSignature")?;
+
+    Ok(object)
+}
+
+/// add partial signature to session
+/// @arguments
+/// - incomplete_session_bytes: incomplete multisig session bytes
+/// - public_key: the other signer's public key
+/// - partial_signature: the other signer's H256 partial signature
+/// @return incompleteSession
+pub fn add_partial_signature(mut ctx: FunctionContext) -> JsResult<JsBuffer> {
+    let incomplete_session_bytes = u8_buffer_argument(&mut ctx, 0)?;
+    let mut session = MultiSigBuilder::from_incomplete(incomplete_session_bytes)
+        .chain_neon(&mut ctx, "Unable to restore the incomplete session")?;
+
+    let public_key = public_key_argument(&mut ctx, 1)?;
+    let partial_signature: H256 = h256_buffer_argument(&mut ctx, 2)?;
+
+    session
+        .add_partial_signature(&public_key, partial_signature)
+        .chain_neon(&mut ctx, "Unable to add partial signature")?;
+
+    let incomplete_session = session.to_incomplete();
+    let mut incomplete_session_buffer= ctx.buffer(incomplete_session.len() as u32)?;
+    ctx.borrow_mut(&mut incomplete_session_buffer, |data| {
+        let slice = data.as_mut_slice();
+        slice.copy_from_slice(&incomplete_session);
+    });
+
+    Ok(incomplete_session_buffer)
+}
+
+/// sign the ready session that gathered all co-signers' signatures
+/// @arguments
+/// - incomplete_session_bytes: incomplete multisig session bytes
+/// @return signature:SchnorrSignature
+pub fn sign(mut ctx: FunctionContext) -> JsResult<JsBuffer> {
+    let incomplete_session_bytes = u8_buffer_argument(&mut ctx, 0)?;
+    let session = MultiSigBuilder::from_incomplete(incomplete_session_bytes)
+        .chain_neon(&mut ctx, "Unable to restore the incomplete session")?;
+
+    let signature = session
+        .signature()
+        .chain_neon(&mut ctx, "Unable to create signature")?
+        .serialize_default()
+        .to_vec();
+
+    let mut signature_buffer= ctx.buffer(signature.len() as u32)?;
+    ctx.borrow_mut(&mut signature_buffer, |data| {
+        let slice = data.as_mut_slice();
+        slice.copy_from_slice(&signature);
+    });
+
+    Ok(signature_buffer)
+}
+
+/// verify the final signature valid or not
+/// @arguments
+/// - signature: final signature Buffer
+/// - message: session H256 message string
+/// - public_keys: all co-signers' public keys
+/// @return boolean
+pub fn verify(mut ctx: FunctionContext) -> JsResult<JsBoolean> {
+    let raw = u8_buffer_argument(&mut ctx, 0)?;
+    let signature = SchnorrSignature::from_default(&raw)
+        .chain_neon(&mut ctx, "Unable to restore signature")?;
+
+    let message = h256_str_argument(&mut ctx, 1)?;
+    let message = Message::from_slice(&message)
+        .chain_neon(&mut ctx, "Unable to form message")?;
+
+    let public_keys = public_key_vector_argument(&mut ctx, 2)?;
```

Comment:
I find the root cause of the `verify` problem. The order of public keys will affect the combined public key.

```suggestion
    let public_keys = public_key_vector_argument(&mut ctx, 2)?;
    public_keys.sort();
```

Reply:
Please also add a test case to cover this problem with these private keys. Thanks!
```
    const buyerRawKeyPair = KeyPair.fromPrivateKey(
        Buffer.from(
            '8b292c13c1bfdec73526669cb121989f92d50fa6d4ad3d60f2c17d21a6a5b8c0',
            'hex',
        ),
    );
    const sellerRawKeyPair = KeyPair.fromPrivateKey(
        Buffer.from(
            'd66f0ecce4e736155a5f80f8229756635c3b060a9538a94ff62a0726f5bd3387',
            'hex',
        ),
    );
```